build/linux/installer/scripts/tomlparser-npm-config.rb

```diff
@@ -0,0 +1,113 @@
+        basic_npm_metrics = parsedConfig[:integrations][:azure_network_policy_manager][:collect_basic_metrics].to_s
+        puts "got:integrations.azure_network_policy_manager.collect_basic_metrics='#{basic_npm_metrics}'"
+        if !basic_npm_metrics.nil? && basic_npm_metrics.strip.casecmp("true") == 0
+            @collect_basic_npm_metrics = true
+        else
+            @collect_basic_npm_metrics = false
+        end
+        puts "set:integrations.azure_network_policy_manager.collect_basic_metrics=#{@collect_basic_npm_metrics}"
+    end
+  rescue => errorStr
+    puts "config::error:Exception while reading config settings for npm basic setting - #{errorStr}, using defaults"
+    @collect_basic_npm_metrics = false
+  end
+end
+
+@configSchemaVersion = ENV["AZMON_AGENT_CFG_SCHEMA_VERSION"]
+puts "****************Start Config Processing********************"
+if !@configSchemaVersion.nil? && !@configSchemaVersion.empty? && @configSchemaVersion.strip.casecmp("v1") == 0 #note v1 is the only supported schema version , so hardcoding it
+  configMapSettings = parseConfigMap
+  if !configMapSettings.nil?
+    populateSettingValuesFromConfigMap(configMapSettings)
+  end
+else
+  if (File.file?(@configMapMountPath))
+    ConfigParseErrorLogger.logError("config::unsupported/missing config schema version - '#{@configSchemaVersion}' , using defaults, please use supported schema version")
+  end
+  @collect_basic_npm_metrics = false
+  @collect_advanced_npm_metrics = false
+end
+
+# Write the settings to file, so that they can be set as environment variables
+file = File.open("integration_npm_config_env_var", "w")
+
+if !file.nil?
+  if @collect_advanced_npm_metrics == true 
+    file.write("export TELEMETRY_NPM_INTEGRATION_METRICS_ADVANCED=1\n")
+    file.write("export AZMON_INTEGRATION_NPM_METRICS_URL_LIST_NODE=#{@npm_node_url}\n")
+    file.write("export AZMON_INTEGRATION_NPM_METRICS_URL_LIST_CLUSTER=#{@npm_cluster_url}\n")
+    file.write("export AZMON_INTEGRATION_NPM_METRICS_DROP_LIST_CLUSTER=\n")
+  elsif @collect_basic_npm_metrics == true
+    file.write("export TELEMETRY_NPM_INTEGRATION_METRICS_BASIC=1\n")
+    file.write("export AZMON_INTEGRATION_NPM_METRICS_URL_LIST_NODE=#{@npm_node_url}\n")
+    file.write("export AZMON_INTEGRATION_NPM_METRICS_URL_LIST_CLUSTER=#{@npm_cluster_url}\n")
+    file.write("export AZMON_INTEGRATION_NPM_METRICS_DROP_LIST_CLUSTER=#{@npm_basic_drop_metrics_cluster}\n")
+  else
+    file.write("export AZMON_INTEGRATION_NPM_METRICS_URL_LIST_NODE=\n")
+    file.write("export AZMON_INTEGRATION_NPM_METRICS_URL_LIST_CLUSTER=\n")
+    file.write("export AZMON_INTEGRATION_NPM_METRICS_DROP_LIST_CLUSTER=\n")
+  end
+  # Close file after writing all environment variables
+  file.close
+else
+  puts "Exception while opening file for writing config environment variables"
+  puts "****************End Config Processing********************"
+end
```

Comment:
minor, but I think, we should have the habit of keeping extra new line so that unix tools process these files as expected.  
https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
its not new line but linefeed at the end of last line. They are applicable/sensitive only for shell scripts/unix scripts not for compiled/interpreted source files.